networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/LibP2PPeer.java

```diff
@@ -41,11 +43,15 @@
 import tech.pegasys.teku.networking.p2p.rpc.RpcRequestHandler;
 import tech.pegasys.teku.networking.p2p.rpc.RpcResponseHandler;
 import tech.pegasys.teku.networking.p2p.rpc.RpcStreamController;
+import tech.pegasys.teku.spec.constants.NetworkConstants;
 
 public class LibP2PPeer implements Peer {
   private static final Logger LOG = LogManager.getLogger();
 
-  private final Map<RpcMethod<?, ?, ?>, RpcHandler<?, ?, ?>> rpcHandlers;
+  private record RpcHandlerAndQutgoingQueue(
```

Comment:
```suggestion
  private record RpcHandlerAndOutgoingQueue(
```
typo

Reply:
Hah, you were first 😄

Reply:
<img width="377" alt="image" src="https://github.com/user-attachments/assets/599515ce-8783-4afe-a0c5-7663a5e40c1c" />
took me ~10s and multiple passes to spot the diff 😅

Reply:
Would be even harder if there wasn't the second Q in Queue 😄

Reply:
nit: May be extract a separate class kind of `PeerThrotling

Reply:
nit: probably make sense to extract a separate class for this, kind of `PeerRpcHandler`? 
TBH I had missed the fact that `RpcHandler` isn't per connection entity in the original PR

Reply:
I took the PeerRpcHandler approach and tested it separately. Think it's nicer yeah, thanks for the suggestion. Feel free to review.